core/camel-core-processor/src/main/java/org/apache/camel/processor/aggregate/AggregateProcessor.java

```diff
@@ -1456,7 +1458,8 @@ public void run() {
                     lock.unlock();
                 }
             }
-
+            recoveryInProgress.set(false);
```

Comment:
Can you move them into the finally block as if an exception was thrown then they are not called